src/components/icon/icon.spec.js

```diff
@@ -514,29 +510,39 @@ describe('MdIcon service', function() {
     });
 
     describe('icon group is not found', function() {
-      it('should log Error', function() {
-        var msg;
-        try {
-          $mdIcon('notfound:someIcon')
-            .catch(function(error) {
-              msg = error.data;
-            });
+      it('should log Error and reject', inject(function($log) {
+        var errorMessage = 'Cannot GET notfoundgroup.svg';
+        var caughtRejection = false;
+
+        $mdIcon('notfound:someIcon')
+          .catch(function(error) {
+            expect(error.data).toBe(errorMessage);
+            caughtRejection = true;
+          });
 
-          $httpBackend.flush();
-        } finally {
-          expect(msg).toEqual('Cannot GET notfoundgroup.svg');
-        }
-      });
+        $httpBackend.flush();
+
+        expect(caughtRejection).toBe(true);
+        expect($log.warn.logs[0]).toEqual([errorMessage]);
```

Comment:
@topherfangio: It isn't related to `$exceptionHandler`. 

`ngMock`'s [`$log` service][1] will keep track of the arguments passed when calling any of its methods. (It also has additional, handy methods for resetting and asserting emptyness).

Although the core (production) `$exceptionHandler` basically delegates to `$log.error()`, `ngMock`'s `$exceptionHandler` doesn't use `$log` at all. It just stores errors in an array (exposed as `$exceptionHandler.errors`) and might also throw if the mode is set to `rethrow` (the default).

[1]: https://docs.angularjs.org/api/ngMock/service/$log